modules/juce_core/native/juce_posix_SharedCode.h

```diff
@@ -1143,24 +1218,22 @@ class ChildProcess::ActiveProcess
     {
         jassert (dest != nullptr && numBytes > 0);
 
-        #ifdef fdopen
-         #error // some crazy 3rd party headers (e.g. zlib) define this function as NULL!
-        #endif
-
-        if (readHandle == nullptr && childPID != 0)
-            readHandle = fdopen (pipeHandle, "r");

```

Comment:
by not doing this we can use use non buffering and non blocking `read` instead of `fread` which can block the process if no output is produced. speed is improved and you won't get character out in chunks with lot of latency and wait time between buffer are filled with new data. it is especially noticeable if you trigger cmake with lots of output from here.